scipy/stats/mstats_basic.py

```diff
@@ -362,7 +362,9 @@ def pearsonr(x,y):
         prob = 0.
     else:
         t_squared = (df / ((1.0 - r) * (1.0 + r))) * r * r
-        prob = betai(0.5*df, 0.5, df/(df + t_squared))
+        tmp = df / (df + t_squared)
+        tmp = np.where(tmp < 1.0, tmp, 1.0)
```

Comment:
My guess, after searching a bit and from what I remember, but haven't verified:
the `betai` here is most likely a complicated way of calculating `stats.t.sf` which would be a lot more informative to use.